engine/runtime/src/test/java/org/enso/interpreter/test/HashCodeTest.java

```diff
@@ -0,0 +1,145 @@
+package org.enso.interpreter.test;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+
+import java.io.ByteArrayOutputStream;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import org.enso.polyglot.MethodNames.Module;
+import org.enso.polyglot.RuntimeOptions;
+import org.graalvm.polyglot.Context;
+import org.graalvm.polyglot.Language;
+import org.graalvm.polyglot.Source;
+import org.graalvm.polyglot.Value;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+public class HashCodeTest {
+  private Context ctx;
+  private Value equalsFunc;
+  private Value hashCodeFunc;
+  private ValuesGenerator valGenerator;
+
+  @Before
+  public void initializeContext() {
+    this.ctx = Context.newBuilder("enso")
+        .allowExperimentalOptions(true)
+        .allowIO(true)
+        .allowAllAccess(true)
+        .logHandler(new ByteArrayOutputStream())
+        .option(
+            RuntimeOptions.LANGUAGE_HOME_OVERRIDE,
+            Paths.get("../../distribution/component").toFile().getAbsolutePath()
+        ).build();
+    final Map<String, Language> langs = ctx.getEngine().getLanguages();
+    assertNotNull("Enso found: " + langs, langs.get("enso"));
+    var src = Source.newBuilder("enso", """
+        equals x y = x == y
+        hashCode x = x.hash_code
+        """, "tmp.enso").buildLiteral();
+    var module = ctx.eval(src);
+    equalsFunc = module.invokeMember(Module.EVAL_EXPRESSION, "equals");
+    hashCodeFunc = module.invokeMember(Module.EVAL_EXPRESSION, "hashCode");
```

Comment:
To avoid exposing `Any.hash_code`, you need to replace the `hashCodeFunc` with direct call to `HashCodeAnyNode` somehow.